gdx-pay-android-googlebilling/src/com/badlogic/gdx/pay/android/googlebilling/PurchaseManagerGoogleBilling.java

```diff
@@ -106,76 +106,79 @@ private String getGlobalSkuTypeFromConfig() {
-                            if (!installationComplete)
-                                observer.handleInstallError(new FetchItemInformationException(
-                                        String.valueOf(responseCode)));
+                            Gdx.app.error(TAG, "onProductDetailsResponse failed, error code is " + responseCode);
+                            if (!installationComplete) {
+                                observer.handleInstallError(new FetchItemInformationException(String.valueOf(responseCode)));
+                            }
 
                         } else {
-                            if (skuDetailsList != null) {
-
-                                for (SkuDetails skuDetails : skuDetailsList) {
-                                    informationMap.put(skuDetails.getSku(), convertSkuDetailsToInformation
-                                            (skuDetails));
-                                    skuDetailsMap.put(skuDetails.getSku(), skuDetails);
-                                }
-                            } else {
-                                Gdx.app.log(TAG, "skuDetailsList is null");
+                            for (ProductDetails productDetails : productDetailsList) {
+                                informationMap.put(productDetails.getProductId(), convertSkuDetailsToInformation
+                                        (productDetails));
+                                productDetailsMap.put(productDetails.getProductId(), productDetails);
                             }
+
                             setInstalledAndNotifyObserver();
                         }
                     }
-                });
+                }
+        );
     }
 
     private String mapOfferType(OfferType type) {
         switch (type) {
             case CONSUMABLE:
             case ENTITLEMENT:
-                return BillingClient.SkuType.INAPP;
+                return BillingClient.ProductType.INAPP;
             case SUBSCRIPTION:
-                return BillingClient.SkuType.SUBS;
+                return BillingClient.ProductType.SUBS;
         }
         throw new IllegalStateException("Unsupported OfferType: " + type);
     }
 
-    private Information convertSkuDetailsToInformation(SkuDetails skuDetails) {
-        String priceString = skuDetails.getPrice();
+    private Information convertSkuDetailsToInformation(ProductDetails productDetails) {
+        String priceString = productDetails.getOneTimePurchaseOfferDetails().getFormattedPrice();
         return Information.newBuilder()
-                .localName(skuDetails.getTitle())
-                .freeTrialPeriod(convertToFreeTrialPeriod(skuDetails.getFreeTrialPeriod()))
-                .localDescription(skuDetails.getDescription())
+                .localName(productDetails.getTitle())
+                //TODO Removed - Dont know if this is applied in some other form
```

Comment:
This is needed for subscriptions. See https://qonversion.io/blog/google-play-billing-library-5-0/ for details

Reply:
Keep in mind that other parts also force CONSUMABLE product type. You may query your SUBSCRIPTION products, offer it, restore, etc.